lib/fs.js

```diff
@@ -1617,8 +1617,15 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (options === undefined)
+    options = {};
+  else if (typeof options === 'string')
+    options = { encoding: options };
+  else if (options === null || typeof options !== 'object')
+    throw new TypeError('options must be a string or an object');
+
   // a little bit bigger buffer and water marks by default
-  options = Object.create(options || {});
+  options = Object.create(options);
```

Comment:
Exactly, now that we don't allow `null`, why do we need `Object.create`? `options` will be an object already.